resharper/resharper-unity/src/UnityVersion.cs

```diff
@@ -40,37 +42,42 @@ public class UnityVersion
         {
             myUnityProjectFileCache = unityProjectFileCache;
             mySolution = solution;
+            myFileSystemTracker = fileSystemTracker;
+            
+            // SolutionDirectory isn't absolute in tests, and will throw if used with FileSystemTracker
+            mySolutionDirectory = solution.SolutionDirectory;
+            if (!mySolutionDirectory.IsAbsolute)
+                mySolutionDirectory = solution.SolutionDirectory.ToAbsolutePath(FileSystemUtil.GetCurrentDirectory());
+            
+            unitySolutionTracker.IsRelatedToUnity.WhenTrue(lifetime, SetActualVersionForSolution);
```

Comment:
This will fire when we have the Unity folder structure and/or Unity references. If we only have a Unity class library, we'll have references and no folder structure, but `SetActualVersionForSolution` only uses the folder structure. We'll end up setting file system watchers on incorrect paths.

I think we should use `UnitySolutionTracker.IsUnityProjectFolder.WhenTrue` for setting up the file system watchers. 

We need to make sure we call `SetActualVersionForSolution` for both Unity projects and Unity class library projects. I think we could remove `IsRelatedToUnity` and just call `SetActualVersionForSolution` in the `IsUnityProjectFolder` handler and set up another handler for `UnityReferencesTracker.HasUnityReference.WhenTrue`.